pkg/osutil/osutil.go

```diff
@@ -279,6 +280,14 @@ func WriteFile(filename string, data []byte) error {
 	return os.WriteFile(filename, data, DefaultFilePerm)
 }
 
+func WriteJSON[T any](filename string, obj T) error {
```

Comment:
Let's merge this separately as well. There are several uses for it in manager code, maybe more.

Reply:
Posted in https://github.com/google/syzkaller/pull/5631.